pkg/sql/schemachanger/rel/query_lang_clauses.go

```diff
@@ -38,7 +38,9 @@ func expanded(c Clauses) Clauses {
 			case and:
 				return true
 			case ruleInvocation:
-				return !cl.rule.isNotJoin
+				if !cl.rule.isNotJoin {
+					return true
+				}
```

Comment:
this should instead be in the penultimate commit

Reply:
Fixed.